test/e2e/lifecycle.go

```diff
@@ -215,3 +219,81 @@ func (c *Client) WaitForResourceReady(namespace, name string, gvr schema.GroupVe
 		return ready.IsTrue(), nil
 	})
 }
+
+// WaitForResourceReady waits until the specified resource in the given namespace are ready.
+func (c *Client) WaitUntilJobDone(namespace, name string) (string, error) {
+	err := wait.PollImmediate(interval, timeout, func() (bool, error) {
+		job, err := c.Kube.Kube.BatchV1().Jobs(namespace).Get(name, metav1.GetOptions{})
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				log.Println(namespace, name, "not found", err)
+				// keep polling
+				return false, nil
+			}
+			return false, err
+		}
+		return operations.IsJobComplete(job), nil
+	})
+	if err != nil {
+		return "", err
+	}
+
+	// poll until the pod is terminated.
+	err = wait.PollImmediate(interval, timeout, func() (bool, error) {
+		pod, err := operations.GetJobPodByJobName(context.TODO(), c.Kube.Kube, namespace, name)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				log.Println(namespace, name, "not found", err)
+				// keep polling
+				return false, nil
+			}
+			return false, err
+		}
+		if pod != nil {
+			for _, cs := range pod.Status.ContainerStatuses {
+				if cs.State.Terminated != nil {
+					return true, nil
+				}
+			}
+		}
+		return false, nil
+	})
+
+	if err != nil {
+		return "", err
+	}
+	pod, err := operations.GetJobPodByJobName(context.TODO(), c.Kube.Kube, namespace, name)
+	if err != nil {
+		return "", err
+	}
+	return operations.GetFirstTerminationMessage(pod), nil
+}
+
+func (c *Client) LogsFor(namespace, name string, gvr schema.GroupVersionResource) (string, error) {
```

Comment:
is gvr actually used for anything? I see you printing the .String, but it's always just pods?